### PR TITLE
Validation particle

### PR DESCRIPTION
**Pull request by ludwigVonKoopa**

### CODE REVIEW

File: src/py_eddy_tracker/dataset/grid.py [hunk 1]
```diff
@@ -2541,6 +2551,11 @@ def get_uv_quad(i0, j0, u, v, m, nb_x=0):
     i1, j1 = i0 + 1, j0 + 1
     if nb_x != 0:
         i1 %= nb_x
+    i_max, j_max = m.shape
```
Comment (AntSimi) [inline src/py_eddy_tracker/dataset/grid.py:2554]: for which use?

File: src/py_eddy_tracker/observations/network.py [hunk 2]
```diff
@@ -97,6 +101,109 @@ def fix_next_previous_obs(next_obs, previous_obs, flag_virtual):
         previous_obs[i_o + 1] = i_o
 
 
+def advect(x, y, c, t0, delta_t):
```
Comment (AntSimi) [inline src/py_eddy_tracker/observations/network.py:104]: Maybe update exemple with this new method/function.
I think those methods must be callable in classic tracking also

File: src/py_eddy_tracker/observations/network.py [hunk 3]
```diff
@@ -109,17 +216,16 @@ def __init__(self, *args, **kwargs):
...
 
-        :param int obs: index of event after the event
-        :param int stopped: index of event before the event
+        :param int obs: index of observation after the event
+        :param int stopped: index of observation before the event
         :param int order: order of relatives accepted
 
         :return: all relative segments
         :rtype: EddiesObservations
         """
```
Comment (AntSimi) [inline src/py_eddy_tracker/observations/network.py:228]: too many whitespace

**Review by AntSimi — CHANGES_REQUESTED** [review]

File: src/py_eddy_tracker/dataset/grid.py [hunk 4]
```diff
@@ -2264,6 +2264,16 @@ def from_netcdf_list(cls, filenames, t, x_name, y_name, indexs=None, heigth=None
             new.datasets.append((t, d))
         return new
 
+    def shift_files(self, t, filename, x_name, y_name, indexs, heigth):
```
Comment (AntSimi) [inline src/py_eddy_tracker/dataset/grid.py:2267]: Maybe add default value to index and height? And maybe kwargs directly to regular grid

File: src/py_eddy_tracker/observations/network.py [hunk 5]
```diff
@@ -97,6 +101,109 @@ def fix_next_previous_obs(next_obs, previous_obs, flag_virtual):
...
 
+def advect(x, y, c, t0, delta_t):
+    """
+    Advect particle from t0 to t0 + delta_t, with data cube.
+
+    :param np.array(float) x: longitude of particles
+    :param np.array(float) y: latitude  of particles
+    :param `~py_eddy_tracker.dataset.grid.GridCollection` c: GridCollection with speed for particles
+    :param int t0: julian day of advection start
+    :param int delta_t: number of days to advect
```
Comment (AntSimi) [inline src/py_eddy_tracker/observations/network.py:112]: Delta t seem not appropriate name

**Review by AntSimi — CHANGES_REQUESTED** [review]

File: src/py_eddy_tracker/dataset/grid.py (hunk 4 above)
Comment (ludwigVonKoopa) [reply src/py_eddy_tracker/dataset/grid.py:2267]: ok for default values, but what do you mean by kwargs directly to regular grid ?

File: src/py_eddy_tracker/observations/network.py (hunk 5 above)
Comment (ludwigVonKoopa) [reply src/py_eddy_tracker/observations/network.py:112]: right

File: src/py_eddy_tracker/dataset/grid.py (hunk 1 above)
Comment (ludwigVonKoopa) [reply src/py_eddy_tracker/dataset/grid.py:2554]: mistake here, i did comment line 2556 & 2557 but they should not be.
Bug here when u,v data are not wide enough to cover all space taken by particles, and particule is advected on edge of u,v matrix. i0 can me > i_max

File: src/py_eddy_tracker/observations/network.py (hunk 2 above)
Comment (ludwigVonKoopa) [reply src/py_eddy_tracker/observations/network.py]: function moved to groups, and example updated

File: src/py_eddy_tracker/observations/network.py (hunk 3 above)
Comment (ludwigVonKoopa) [reply src/py_eddy_tracker/observations/network.py:124]: right

File: src/py_eddy_tracker/dataset/grid.py (hunk 4 above)
Comment (AntSimi) [reply src/py_eddy_tracker/dataset/grid.py]: Lot of those arguments are not used by method just transfer to another one, so maybe we could **kwargs to be transparent

**Review by AntSimi — CHANGES_REQUESTED** [review]
When you modify example, you may run notebooks/README.md to update ipynb for binder use

**Comment by AntSimi** [comment]
https://github.com/AntSimi/py-eddy-tracker/pull/87/checks#step:6:22, look here there are some new warning

File: src/py_eddy_tracker/dataset/grid.py (hunk 4 above)
Comment (ludwigVonKoopa) [reply src/py_eddy_tracker/dataset/grid.py]: ok, done

**Review by AntSimi — APPROVED** [review]

**Pull request merged**